### PR TITLE
Adapt UI based on the person accesing the page.

### PR DESCRIPTION
**Pull request by iooeduard**
'isFinancial' and 'isFreak' functions are yet to be implemented on the smart contract side.

### CODE REVIEW

File: packages/react-app/src/components/Roles/roles.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const contract = useContext(ContractContext);
+  const address = Web3Service.getAddress();
+  const financial = EthContract.isFinancial(contract, address);
+  const freak = EthContract.isFreak(contract, address);
+
+  function isGuest() {
+    return !financial && !freak;
+  }
+
+  return (
```
Comment (dasilvaluis) [inline packages/react-app/src/components/Roles/roles.jsx:21]: I realized that we could have done something like, but I over-engineered a little bit :)
Up to you if you prefer this approach.

```js
const rolesPages = [
  {
    matcher: contract.isFinancial,
    getPage: () => <FinancialRole />
  },
  {
    matcher: contract.isFreak,
    getPage: () => <FreakRole />
  },
  {
    matcher: (address) => !contract.isFreak(address) && !contract.isFinancial(address),
    getPage: () => <FreakRole />
  }
];

const page = rolesPages.find(({ matcher }) => matcher(address))?.getPage();

return page
```

It this latest you would need neither the RoleRouter or the RoleRoute components

File: packages/react-app/src/App.jsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ function App() {
       </Header>
       <Body>
         <ContractProvider>
-          <FinancialRole />
+          <RoleComponents />
```
Comment (dasilvaluis) [inline packages/react-app/src/App.jsx]: I suggest something more meaningful

```suggestion
          <RolePageSwitcher />
```

**Review by dasilvaluis — CHANGES_REQUESTED** [review]

File: packages/react-app/src/components/Roles/roles.jsx (hunk 1 above)
Comment (iooeduard) [reply packages/react-app/src/components/Roles/roles.jsx:21]: Looks super clean but I like the other approach more since I fully understood it.

**Review by dasilvaluis — APPROVED** [review]

**Comment by iooeduard** [comment]
[UI should adapt to the role of the caller](https://trello.com/c/Ecvwex6K/205-ui-should-adapt-to-the-role-of-the-caller)

**Pull request merged**